astropy/units/function/core.py

```diff
@@ -615,6 +622,15 @@ def __lt__(self, other):
     def __le__(self, other):
         return self._comparison(other, self.value.__le__)
 
+    def __lshift__(self, other):
+        """Unit converstion operator `<<`"""
+        try:
+            other = Unit(other, parse_strict='silent')
+        except UnitTypeError:
+            return NotImplemented
+
+        return self.__class__(self, other, copy=False, subok=True)
+
```

Comment:
I'm having trouble groking how `ilshift` would work in this context. Might be worth waiting on this one, unless you can give me an example test?

Reply:
It would be something like:
```
    try:
        other = self._unit_class(other)
    except UnitConversionError:
        return NotImplemented

    try:
        physical_factor = self.physical_unit.to(other.physical_unit)
    except UnitConversionError:
        # Maybe via equivalencies?  Now we do make a temporary copy.
        try:
            value = self._to_value(other)
        except UnitConversionError:
            return NotImplemented

        self.view(np.ndarray)[...] = value
    else:
        self.view(np.ndarray)[...] += self.from_physical(factor)
```
(Note: *not* tested...)

Reply:
I implemented `LogQuantity.__ilshift__` in the latest push, following your suggestion further up. However, I needed to also catch `ValueError` twice (explicitly: `ValueError: Unit mag(erg / (cm2 Hz s)) is not a physical unit`), which sends it into the `_to_value` branch, so I'm not sure that it's actually doing "in-place" lshift in the test.

Reply:
I checked and it indeed doesn't do it in-place. Will try to find a fix and push directly to your branch, so we can get this in...